pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -963,10 +970,29 @@ void updateIdealStateOnSegmentCompletion(String realtimeTableName, String commit
     }, RetryPolicies.exponentialBackoffRetryPolicy(10, 1000L, 1.2f));
   }
 
-  private boolean isTablePaused(IdealState idealState) {
+  public static boolean isTablePaused(IdealState idealState) {
+    PauseState pauseStatus = extractTablePauseStatus(idealState);
+    if (pauseStatus != null) {
+      return pauseStatus.isPaused();
+    }
+    // backward compatibility
+    // TODO : remove this handling after next release.
+    //  Expectation is that all table IS are migrated to the newer representation.
     return Boolean.parseBoolean(idealState.getRecord().getSimpleField(IS_TABLE_PAUSED));
   }
 
+  private static PauseState extractTablePauseStatus(IdealState idealState) {
+    String pauseStatusStr = idealState.getRecord().getSimpleField(PinotLLCRealtimeSegmentManager.PAUSE_STATUS);
```

Comment:
Ditto: pauseStateStr?
Could you please update all instances of "pause status" to "pause state"?

Reply:
Both in code and the comments.